recipes/systemc/meta.yaml

```diff
@@ -0,0 +1,146 @@
+{% set name = "systemc" %}
+{% set version = "2.3.4" %}
+{% set major_minor = ".".join(version.split(".")[0:2]) %}
+
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/accellera-official/systemc/archive/refs/tags/{{ version }}.tar.gz
+  sha256: bfb309485a8ad35a08ee78827d1647a451ec5455767b25136e74522a6f41e0ea
+  patches:
+    - patches/0001-examples-sysc-2.3-simple_async-needs-PTHREADS.patch  # https://github.com/accellera-official/systemc/pull/44
+    - patches/0002-rm-project-default-CXX-std.patch
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - make      # [not win]
+    - autoconf  # [not win]
+    - automake  # [not win]
+    - libtool   # [not win]
+    - cmake
+
+# NOTE windows cmake build doesn't include the examples in it's install
+test:                         # [win]
+  source_files:               # [win]
+    - examples                # [win]
+  requires:                   # [win]
+    - {{ compiler('cxx') }}   # [win]
+  commands:                   # [win]
+    - cd examples/build-msvc  # [win]
+    - nmake /nologo CONFIG=Release PLATFORM=x64 VCBUILD=vcbuild check  # [win64]
+    - nmake /nologo CONFIG=Release PLATFORM=Win32 VCBUILD=vcbuild check  # [win32]
+
+# split unix packages into:
+# libsystemc dynamic-library
+# systemc-dev package depends on libsystemc and adds header and unversioned .so symlink
+# systemc-static just the static library
+# systemc-doc  documentation including the installed examples
+
+outputs:                                                         # [not win]
```

Comment:
Thanks for your patience. Why not split the Windows package too?

Reply:
System-C doesn't support shared-library builds on Windows and also doesn't include the documentation (need to double check that).  I put a comment about the shared library support not working in https://github.com/conda-forge/staged-recipes/pull/22280/files#diff-b8d2b920295573b173d5ed931eae74bca220fad3f892fc99d644f20d84debd9aR9-R11 Perhaps I should also explain that here.

Without shared library or docs, splitting the package didn't make sense to me (although I agree that it is ugly).

Reply:
Ideally we want consistency so that downstream packages don't need to use platform selectors to build correctly against this package. Maybe I'm wrong, but it seems like unix packages would build against "systemc-dev", but windows would build against "systemc"?

Reply:
That's a very good point.  I think in this case, it may make sense for the windows package to ship `systemc-dev` that downstreams would build against and possibly a `systemc` metapackage that on windows only depends on `systemc-dev`?

Reply:
I think the outputs for unix and windows should be according to this chart

systemc        both
systemc-dev    both
systemc-doc    both
libsystemc     unix-only because static links for windows